### PR TITLE
MOEN-8794 MOEN-9805 MOEN-9804 version 11 upgrade

### PR DESCRIPTION
**Pull request by arshiya-moengage**
Android:
* MOEN-8794: Flutter set method call handler to null when the engine is in detached state
* MOEN-9805: Update Flutter Inbox Plugin with SDK version v6
* MOEN-9804: Update Flutter Core Plugin with native SDK v11

Common:
* Added Push Token callback

### CODE REVIEW

File: core/lib/utils.dart [hunk 1]
```diff
@@ -159,3 +160,25 @@ PushCampaign pushCampaignFromJson(dynamic methodCallArgs) {
 
   return null;
 }
+
+PushToken pushTokenFromJson(dynamic methodCallArgs) {
```
Comment (umangmoe) [inline core/lib/utils.dart:164]: See if we can avoid nulls with default values or something else. Dart has introduced null safety support similar to Kotlin and might be a problem if the client enables null safety.

File: core/android/build.gradle [hunk 2]
```diff
@@ -16,6 +16,7 @@ buildscript {
 
 rootProject.allprojects {
     repositories {
+        mavenLocal()
```
Comment (umangmoe) [inline core/android/build.gradle:19]: remove this.

**Review by umangmoe — CHANGES_REQUESTED** [review]

**Comment by umangmoe** [comment]
@arshiya-moengage  Add API similar to the one here - https://github.com/moengage/MoEngage-Unity-SDK/blob/master/Android/android-unity-wrapper/unity-wrapper/src/main/java/com/moengage/unity/wrapper/MoEInitializer.java#L52

File: core/lib/push_token.dart [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import 'package:moengage_flutter/constants.dart';
+
+class PushToken {
+  String platform;
+  String token;
+  String pushService;
```
Comment (umangmoe) [inline core/lib/push_token.dart:6]: Use enum

File: inbox/android/gradle/wrapper/gradle-wrapper.properties [hunk 4]
```diff
@@ -2,4 +2,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.5-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.5-bin.zip
```
Comment (umangmoe) [inline inbox/android/gradle/wrapper/gradle-wrapper.properties:5]: any reason this was changed from `all` to `bin`

File: core/lib/push_token.dart (hunk 3 above)
Comment (arshiya-moengage) [reply core/lib/push_token.dart]: done

File: inbox/android/gradle/wrapper/gradle-wrapper.properties (hunk 4 above)
Comment (arshiya-moengage) [reply inbox/android/gradle/wrapper/gradle-wrapper.properties]: Got overwritten while rebasing, fixed now.

**Review by umangmoe — APPROVED** [review]

**Pull request merged**